src/plugins/irc-events/connection.js

```diff
@@ -0,0 +1,75 @@
+var _ = require("lodash");
+var identd = require("../../identd");
+var Msg = require("../../models/msg");
+
+module.exports = function(irc, network) {
+	var client = this;
+
+	client.emit("msg", {
+		chan: network.channels[0].id,
+		msg: new Msg({
+			text: "Network created, connecting to " + network.host + ":" + network.port + "..."
+		})
+	});
+
+	irc.on("raw socket connected", function() {
+		identd.hook(irc.connection.socket, network.username);
```

Comment:
So in practice, the `irc.connection.socket.socket` property gets used by the `hook` method?!